sentry_streams/src/run.rs

```diff
@@ -0,0 +1,99 @@
+use std::ffi::OsString;
+
+use clap::Parser;
+use pyo3::prelude::*;
+
+use crate::utils::traced_with_gil;
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    #[command(flatten)]
+    pub py_config: PyConfig,
+
+    #[command(flatten)]
+    pub runtime_config: RuntimeConfig,
+}
+
+#[derive(Parser, Debug)]
+pub struct PyConfig {
+    /// Path of the python interpreter executable
+    #[arg(short, long)]
+    pub exec_path: Option<OsString>,
+
+    #[arg(short, long)]
+    /// Paths to the relevant python modules
+    pub module_paths: Option<Vec<OsString>>,
+}
+
+#[derive(Parser, Debug)]
+pub struct RuntimeConfig {
+    /// The name of the adapter
+    #[arg(short, long)]
+    pub adapter_name: String,
+
+    /// The deployment config file path. Each config file currently corresponds to a specific pipeline.
+    #[arg(short, long)]
+    pub config_file: OsString,
+
+    /// The segment id to run the pipeline for
+    #[arg(short, long)]
+    pub segment_id: Option<String>,
+
+    /// The name of the application
+    pub application_name: String,
+}
+
+pub fn run(args: Args) -> Result<(), Box<dyn std::error::Error>> {
+    pyo3::prepare_freethreaded_python();
+
+    let Args {
+        py_config,
+        runtime_config,
+    } = args;
+
+    traced_with_gil!(|py| -> PyResult<()> {
+        if let Some(exec_path) = py_config.exec_path {
+            PyModule::import(py, "sys")?.setattr("executable", exec_path)?;
+        }
+        if let Some(module_paths) = py_config.module_paths {
+            PyModule::import(py, "sys")?.setattr("path", module_paths)?;
+        }
+        Ok(())
+    })?;
+
+    let runtime: Py<PyAny> = traced_with_gil!(|py| {
+        let runtime = py
+            .import("sentry_streams.runner")?
+            .getattr("load_runtime")?
+            .call1((
+                runtime_config.adapter_name,
+                runtime_config.config_file,
+                runtime_config.segment_id,
+                runtime_config.application_name,
+            ))?
+            .unbind();
+
+        {
+            let runtime = runtime.clone_ref(py);
+            ctrlc::set_handler(move || {
+                traced_with_gil!(|py| {
+                    runtime
+                        .bind(py)
+                        .call_method0("shutdown")
```

Comment:
I don't think this will work.

```
    def shutdown(self) -> None:
        """
        Shutdown the arroyo processors allowing them to terminate the inflight
        work.
        """
        raise NotImplementedError
```

While the signal management code remained in the python version the signal handler that actually takes care of terminating the consumer is in `consumer.rs`

Reply:
IT works because rust also register the signal handler and python signal handler is never reached as the rust code intercepts it. 
I forgot to remove the python code when I added the rust implementation. I had tried to do it inpython first and could not figure it out. It worked fine in rust